src/test/java/edu/ucla/library/prl/harvester/services/HarvestJobSchedulerServiceIT.java

```diff
@@ -169,29 +169,33 @@ public void testInstantiationExistingJob(final Vertx aVertx, final VertxTestCont
 
         // Add jobs before instantiation of the service
-        addInstitution().compose(institutionID -> addJob(institutionID, List.of(SET1, SET2))).compose(initialJob -> {
+        addInstitution().compose(institutionID -> addJob(institutionID, List.of(SET1, SET2))).compose(jobID -> {
             final Checkpoint jobResultReceived = aContext.checkpoint();
 
             LOGGER.debug(MessageCodes.PRL_030);
 
             aVertx.eventBus().<JsonObject>consumer(HarvestJobSchedulerService.JOB_RESULT_ADDRESS, message -> {
-                final JobResult jobResult = new JobResult(message.body());
-                final CompositeFuture queryBackingServices =
-                        CompositeFuture.all(myHarvestScheduleStoreServiceProxy.getJob(jobResult.getJobID()),
-                                TestUtils.getAllDocuments(mySolrClient));
-
-                queryBackingServices.onSuccess(results -> {
-                    final Job job = results.resultAt(0);
-                    final SolrDocumentList solrDocs = results.resultAt(1);
-
-                    aContext.verify(() -> {
-                        assertEquals(SET1_RECORD_COUNT + SET2_RECORD_COUNT, jobResult.getRecordCount());
-                        assertEquals(jobResult.getRecordCount(), solrDocs.getNumFound());
-                        assertTrue(job.getLastSuccessfulRun().isPresent());
-                        assertEquals(jobResult.getStartTime().withNano(0).toInstant(),
-                                job.getLastSuccessfulRun().get().withNano(0).toInstant());
-
-                        jobResultReceived.flag();
+                // The application code that updates the database is listening on this address too, so wait a second to
+                // ensure that happened
+                aVertx.setTimer(1000, timerID -> {
+                    final JobResult jobResult = new JobResult(message.body());
+                    final CompositeFuture queryBackingServices =
+                            CompositeFuture.all(myHarvestScheduleStoreServiceProxy.getJob(jobResult.getJobID()),
+                                    TestUtils.getAllDocuments(mySolrClient));
+
+                    queryBackingServices.onSuccess(results -> {
+                        final Job job = results.resultAt(0);
+                        final SolrDocumentList solrDocs = results.resultAt(1);
+
+                        aContext.verify(() -> {
+                            assertEquals(SET1_RECORD_COUNT + SET2_RECORD_COUNT, jobResult.getRecordCount());
+                            assertEquals(jobResult.getRecordCount(), solrDocs.getNumFound());
+                            assertTrue(job.getLastSuccessfulRun().isPresent());
+                            assertEquals(jobResult.getStartTime().withNano(0).toInstant(),
+                                    job.getLastSuccessfulRun().get().withNano(0).toInstant());
+
+                            jobResultReceived.flag();
+                        });
```

Comment:
The change here is simply wrapping the backing service queries in Vertx#setTimer to delay their execution.

Reply:
To clarify my understanding, this delay is because the database is updated independently of this next action, so the pause gives time for that to happen? Would it be desirable to have the database result be the trigger for the next step instead (if I've understood correctly)?

Reply:
Yes, that first part is what I was getting at... if one _has_ to be done before the other, isn't there a coupling there (and in which case I'm wondering if the timer approach is brittle -- we've done this in the past and that's usually where we see issues in the future).

I'm not sure I'm following the second part, probably because I don't have my head wrapped around all this very well yet... It sounds like you're saying the coupling is only a result of the tests (not something that's important in the application code). In that case, I'd wonder why the tests have the coupling if the application doesn't. Is this a result of the way the tests are written: to use multiple parts of the API rather than more atomically testing individual aspects?

Reply:
I guess _decoupling_ wasn't the right term, but _loosening_ the coupling (so now rather than having a reference to the database service itself, the message send happens by coordinating production/consumption on the event bus address).

Sure, in the running application we _do_ want e.g. adding a new job row to the database to coincide with adding a new cron trigger to the in-memory scheduler, but it's more in line with separation of concerns to have each service independent in-and-of-themselves, and compose them at the handler level.

I'm not aware of the timer approach "brittleness", but am curious to see examples from the past to determine if that applies here as well.

Re: the 2nd paragraph -- prior to the "decoupling" commit (which I should have called "loosening the coupling"), the only consumer of the `JOB_RESULT_ADDRESS` was the test code. And I think that that was/is a bad practice. It's not the coupling that I consider the problem, it's the manner and location of that coupling which was the problem.

Reply:
The previous brittleness issues were with Bucketeer and I believe we've removed all those types of tests now, preferring instead to have a chain of events when things are actually interdependent (loosely or tightly) -- I think generally our tests are getting better. This was from before you were on the Services team, though. There was a period where we seemed to always be adjusting timeouts to give processes that had changed in some way (or had the systems underneath them change in some way) a little more time to complete (or trying to debug problems that were surfacing that looked like timeout issues, when there was something deeper going on). I just wanted to raise the concern. I also wasn't raising the issue of coupling itself as much as of using a timer to support that (as the connection between the two activities).